src/ophyd_async/core/_signal.py

```diff
@@ -504,13 +528,19 @@ def queue_value(value: SignalDatatypeT, signal=signal):
 
     if done_status is not None:
         done_status.add_callback(q.put_nowait)
-
+    overall_deadline = (
+        time.monotonic() + except_after_time if except_after_time else None
+    )
     try:
         while True:
-            # yield here in case something else is filling the queue
-            # like in test_observe_value_times_out_with_no_external_task()
-            await asyncio.sleep(0)
-            item = await get_value()
+            if overall_deadline and time.monotonic() >= overall_deadline:
+                raise asyncio.TimeoutError(
+                    f"observe_value was still observing signals "
+                    f"{[signal.source for signal in signals]} after "
+                    f"timeout {except_after_time}s"
+                )
+
+            item = await get_value(_get_iteration_timeout(timeout, overall_deadline))
```

Comment:
`asyncio.wait_for` accepts timeout=None without scheduling a task, so I would just use that...
```suggestion
            iteration_timeout = _get_iteration_timeout(timeout, overall_deadline)
            item = await asyncio.wait_for(q.get(), iteration_timeout)
```